modelforge/potential/models.py

```diff
@@ -551,124 +546,114 @@ def apply_bwd(res, grads):
+        ).to(box_vectors.device)
 
+        self.periodic = periodic
 
-class NeuralNetworkPotentialFactory:
-    """
-    Factory class for creating instances of neural network potentials for training/inference.
-    """
+    def forward(self, coordinate_i: torch.Tensor, coordinate_j: torch.Tensor):
+        r_ij = coordinate_i - coordinate_j
 
-    @staticmethod
-    def generate_potential(
-        *,
-        use: Literal["training", "inference"],
-        potential_parameter: Union[
-            ANI2xParameters,
-            SAKEParameters,
-            SchNetParameters,
-            PhysNetParameters,
-            PaiNNParameters,
-            TensorNetParameters,
-        ],
-        runtime_parameter: Optional[RuntimeParameters] = None,
-        training_parameter: Optional[TrainingParameters] = None,
-        dataset_parameter: Optional[DatasetParameters] = None,
-        dataset_statistic: Optional[Dict[str, float]] = None,
-        potential_seed: Optional[int] = None,
-        simulation_environment: Optional[Literal["PyTorch", "JAX"]] = None,
-    ) -> Union[Type[torch.nn.Module], Type[JAXModel], Type[pl.LightningModule]]:
+        if self.periodic:
+            r_ij = (
+                torch.remainder(r_ij + self.box_lengths / 2, self.box_lengths)
+                - self.box_lengths / 2
+            )
+
+        d_ij = torch.norm(r_ij, dim=1, keepdim=True, p=2)
+        return r_ij, d_ij
+
+
+class NeighborlistForInferenceNonUniquePairs(torch.nn.Module):
+
+    def __init__(self, cutoff: float, displacement_function: Displacement):
         """
-        Creates an NNP instance of the specified type, configured either for training or inference.
+        Initialize the ComputeInteractingAtomPairs module.
 
         Parameters
         ----------
-        use : Literal["training", "inference"]
-            The use case for the model instance, either 'training' or 'inference'.
-        potential_parameter : Union[ANI2xParameters, SAKEParameters, SchNetParameters, PhysNetParameters, PaiNNParameters, TensorNetParameters]
-            Parameters specific to the potential.
-        runtime_parameter : Optional[RuntimeParameters], optional
-            Parameters for configuring the runtime environment.
-        training_parameter : Optional[TrainingParameters], optional
-            Parameters for configuring the training.
-        dataset_parameter : Optional[DatasetParameters], optional
-            Parameters for configuring the dataset.
-        dataset_statistic : Optional[Dict[str, float]], optional
-            Statistics of the dataset for normalization purposes.
-        potential_seed : Optional[int], optional
-            Seed for the random number generator.
-        simulation_environment : Optional[Literal["PyTorch", "JAX"]], optional, None
-            The simulation environment to use for training/inference. Will override the runtime parameter if provided.
+        cutoffs : Dict[str, unit.Quantity]
```

Comment:
I think we need to update this doc string to be Dict[str, float].   I don't think the neighbor list needs units, as it would not be something a user would directly access.

Reply:
Also I'm seeing the multiple pairlists have been removed from this PR. Is there a reason why?